File: leetcode/929-unique-email-address.py
class Solution:
    def numUniqueEmails(self, emails):
        """
        :type emails: List[str]
        :rtype: int
        """
        
        # remove .
        for i, e in enumerate(emails):
          es = e.split('@')
          es[0] = es[0].replace('.', '').split('+')[0]
          emails[i] = "@".join(es)
                  
        return len(set(emails))

File: leetcode/929-unique-email-address.go
package leetcode 

import "strings"

func numUniqueEmails(emails []string) int {
  mp := make(map[string]struct{})
  for _, e := range emails {
    es := strings.Split(e, "@")
    es[0] = strings.Split(strings.Replace(es[0],".", "", -1), "+")[0]
    mp[strings.Join(es, "@")] = struct{}{}
  }

  return len(mp)
}
